graphql_compiler/tests/schema_transformation_tests/test_merge_schemas.py

```diff
@@ -456,23 +479,755 @@ def test_clashing_directives(self):
+                OrderedDict([
+                    ('first', parse(ISS.basic_schema)),
+                    ('second', parse(ISS.same_field_schema)),
+                ]),
+                [
+                    CrossSchemaEdgeDescriptor(
+                        edge_name = 'example_edge',
+                        outbound_side = FieldReference(
+                            schema_id = 'first',
+                            type_name = 'String',
+                            field_name = 'id',
+                        ),
+                        inbound_side = FieldReference(
+                            schema_id = 'second',
+                            type_name = 'Droid',
+                            field_name = 'identifier',
+                        ),
+                        out_edge_only=False,
+                    ),
+                ]
+            )
+
+    def test_invalid_edge_enum_type(self):
+        with self.assertRaises(InvalidCrossSchemaEdgeError):
+            merge_schemas(
+                OrderedDict([
+                    ('first', parse(ISS.basic_schema)),
+                    ('second', parse(ISS.enum_schema)),
+                ]),
+                [
+                    CrossSchemaEdgeDescriptor(
+                        edge_name = 'example_edge',
+                        outbound_side = FieldReference(
+                            schema_id = 'first',
+                            type_name = 'Human',
+                            field_name = 'id',
+                        ),
+                        inbound_side = FieldReference(
+                            schema_id = 'second',
+                            type_name = 'Height',
+                            field_name = 'identifier',
+                        ),
+                        out_edge_only=False,
+                    ),
+                ]
+            )
+
+    def test_invalid_edge_nonexistent_field(self):
+        with self.assertRaises(InvalidCrossSchemaEdgeError):
+            merge_schemas(
+                OrderedDict([
+                    ('first', parse(ISS.basic_schema)),
+                    ('second', parse(ISS.same_field_schema)),
+                ]),
+                [
+                    CrossSchemaEdgeDescriptor(
+                        edge_name = 'example_edge',
+                        outbound_side = FieldReference(
+                            schema_id = 'first',
+                            type_name = 'Human',
+                            field_name = 'id',
+                        ),
+                        inbound_side = FieldReference(
+                            schema_id = 'second',
+                            type_name = 'Person',
+                            field_name = 'name',
+                        ),
+                        out_edge_only=False,
+                    ),
+                ]
+            )
+
+    def test_invalid_edge_clash_with_existing_field(self):
+        clashing_field_schema = dedent('''\
```

Comment:
same question as above, https://github.com/kensho-technologies/graphql-compiler/pull/430/files#r306861443

Reply:
This confusion is probably due to poor naming, since two different things are both called fields here: one, the two ends of the edge specified in the `@stitch` directive ('id', 'identifier', etc), which is how it's used in test_invalid_edge_nonexistent_field; two, the newly added fields that describe the edge ('out_example_edge', etc), which is how it's used in test_invalid_edge_clash_with_existing_field.

I think this can be fixed by calling the first case 'property field' and the second case 'vertex field'. Will change.